include/net/net_ip.h

```diff
@@ -79,27 +79,19 @@ enum net_sock_type {
 /** IPv6 address structure */
 struct in6_addr {
 	union {
-		u8_t		u6_addr8[16];
-		u16_t		u6_addr16[8]; /* In big endian */
-		u32_t		u6_addr32[4]; /* In big endian */
-	} in6_u;
-#define s6_addr			in6_u.u6_addr8
-#define s6_addr16		in6_u.u6_addr16
-#define s6_addr32		in6_u.u6_addr32
+		u8_t s6_addr[16];
+		u16_t s6_addr16[8]; /* In big endian */
+		u32_t s6_addr32[4]; /* In big endian */
+	};
 };
 
 /** IPv4 address */
 struct in_addr {
 	union {
-		u8_t		u4_addr8[4];
-		u16_t		u4_addr16[2]; /* In big endian */
-		u32_t		u4_addr32[1]; /* In big endian */
-	} in4_u;
-#define s4_addr			in4_u.u4_addr8
-#define s4_addr16		in4_u.u4_addr16
-#define s4_addr32		in4_u.u4_addr32
-
-#define s_addr			s4_addr32[0]
+		u8_t s_addr8[4];
```

Comment:
I think we could be consistent here. In Linux there was no s4_addr* defines so I added them here so that it would be consistent with IPv6 version.
So what about having here
s4_addr
s4_addr16
s4_addr32
and also
s_addr
which is actual used in regular Linux headers. This way we do not need to change anything in the code either. In ancient times the s_addr was used because IPv4 was the "only" protocol there was. As IPv6 came along, then the s6_* variants were used in that side. To me it looks like that having s4_* variants would make sense here.

Reply:
So, per POSIX conventions, IPv4 is the "default" thing, types/structures used by it don't have any digits. It's IPv6 types/structures which get "6" infix. POSIX itself http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/netinet_in.h.html says:

> the in_addr structure, which shall include at least the following member:
> in_addr_t  s_addr

> the in6_addr structure, which shall include at least the following member:

> uint8_t s6_addr[16]

I.e. POSIX itself doesn't define more members beyond s_addr and s6_addr, but doesn't preclude them either. Modern version of Linux defines extra members just for s6_addr (s6_addr16, s6_addr32). We want to define extra member-aliases for s_addr, fine, but we should follow POSIX conventions, i.e. use "s_addr" as the prefix, because "s4_addr" stands out of place (yes, it's more understandable, but it doesn't follow convention, which can only lead to confusion). So, I think @rlubos' changes are for good and clean up the situation.

Reply:
Ok, so we want to keep it as it was. I'm fine with that, the reasoning makes sense. I'm happy as long as we get rid of those  defines :)

Reply:
I'm +1 to have this change in. @jukkar, do you buy the "POSIX naming convention consistency" argument, or want `s4_*` back?

Reply:
I would like to keep the s4_* versions, of course we need also the s_addr too.